### PR TITLE
Integrate GitHub Actions

### PR DESCRIPTION
**Pull request by Tarang74**
Changes made:
- Removed husky pre-commit hooks and commit convention linting due to compatibility issues.
- Added 2 GitHub Actions workflows 

Workflows:
- For `api/`:
  - Run `cargo build` & `cargo test` (feel free to suggest additional workflows)

- For `client/`:
  - Refactoring `client/app/` and `client/components/` (linting/formatting) 
  - Exporting Rust types to TypeScript via `rust-types-to-json` project
  
These workflows activate on every PR. If additional commits are made to an existing PR, a new job will begin on the branches' latest commit.

Note: The two TypeScript workflows have the ability to push new commits to the PR.

### CODE REVIEW

**Comment by Tarang74** [comment]
@QUT-IT-Chads/admin 

Please `squash and merge` this PR. The intermediate commits are not important.

File: client/app/review/create/(components)/(Form)/Form.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+import { NewReview } from 'types/new_review';
+import TextArea from '@components/UI/TextArea';
+
+import styles from './Form.module.scss';
+import Dropdown from '@components/UI/Dropdown';
+import { Option } from 'react-dropdown';
+
+async function sendFormData(newReview: NewReview) {
+    await fetch('http://localhost:8000/api/review/create', {
```
Comment (BrookJeynes) [inline client/app/review/create/(components)/(Form)/Form.tsx]: Don't hardcode the localhost section, either store this in a global variable or the .env file.

```ts
// src/config.ts

interface Config {
    QutReviewApiUrl: string
}

const dev = {
    // just have these as empty strings as we don't have a staging or prod api just yet
    QutReviewApiUrl: "https://staging.api"
}

const prod = {
    // just have these as empty strings as we don't have a staging or prod api just yet
    QutReviewApiUrl: "https://prod.api"
}

const config = process.env.REACT_APP_STAGE === "prod" ? prod : process.env.REACT_APP_STAGE === "dev" ? dev : {}

export default {
    QutReviewApiUrl: "http://localhost:8000",
    ...config
} as Config;
```

```ts
// package.json

"scripts": {
    "start": "cross-env REACT_APP_STAGE=local && react-scripts start",
    "build": "cross-env REACT_APP_STAGE=local && react-scripts build",
    "build:dev": "cross-env REACT_APP_STAGE=dev && react-scripts start",
    "build:prod": "cross-env REACT_APP_STAGE=prod && react-scripts build",
    ...
}
```

File: client/app/review/create/(components)/(Form)/Form.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        .then((res) => res.json())
+        .then((data) => {
+            console.log(data);
+        });
+}
+
+export default function Form() {
+    const [reviewBody, setReviewBody] = useState('');
+
+    const gradeOptions: Option[] = [
```
Comment (BrookJeynes) [inline client/app/review/create/(components)/(Form)/Form.tsx]: I'm not sure but will other parts of the code need this variable? if so we should make it global so everywhere can access it, or make it an export

File: client/app/review/create/(components)/(Form)/Form.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                name="reviewGrade"
+                options={gradeOptions}
+                default={defaultGrade}
+                placeholder="Select a grade"
+                required={true}
+            />
+            <button
+                type="submit"
+                className={styles.submitButton}
+                onClick={async (e) => {
```
Comment (BrookJeynes) [inline client/app/review/create/(components)/(Form)/Form.tsx]: This file is quite small, maybe move the onClick functionality to a function above to clean up the jsx section?

File: client/app/review/create/page.tsx [hunk 4]
```diff
@@ -1,3 +1,16 @@
...
+import Form from './(components)/(Form)';
+
 export default function Page() {
-    return <div>Hello world!</div>;
+    return (
+        <div className={styles.formContainer}>
+            <div className={styles.formHeading}>
+                <h1 className={styles.unitCode}>CAB202</h1>
+                <h2 className={styles.unitName}>
+                    Microprocessors and Digital Systems
```
Comment (BrookJeynes) [inline client/app/review/create/page.tsx]: Is this hard-coded here on purpose?

File: client/components/UI/Dropdown/Dropdown.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+export default function Dropdown(props: DropdownProps) {
+    return (
+        <FormItem id={props.id} label={props.label} required={props.required}>
+            <ReactDropdown
+                options={props.options}
+                value={props.default}
+                placeholder={props.placeholder}
+            />
+            {/* <select className={styles.dropdownSelect} id={props.id}>
```
Comment (BrookJeynes) [inline client/components/UI/Dropdown/Dropdown.tsx]: Try not to commit commented out code, if it's necessary then leave a comment as to why it's commented out

**Review by BrookJeynes — CHANGES_REQUESTED** [review]

File: client/app/review/create/(components)/(Form)/Form.tsx (hunk 1 above)
Comment (Tarang74) [reply client/app/review/create/(components)/(Form)/Form.tsx]: Great suggestion

File: client/app/review/create/page.tsx (hunk 4 above)
Comment (Tarang74) [reply client/app/review/create/page.tsx]: Yes just for now until the routes are setup

File: client/components/UI/Dropdown/Dropdown.tsx (hunk 5 above)
Comment (Tarang74) [reply client/components/UI/Dropdown/Dropdown.tsx]: This code was not meant to be committed with this branch, I’ll remove it from here and make the changes in the proper branch

File: client/app/review/create/(components)/(Form)/Form.tsx (hunk 3 above)
Comment (Tarang74) [reply client/app/review/create/(components)/(Form)/Form.tsx]: Sure that’s totally doable

File: client/app/review/create/(components)/(Form)/Form.tsx (hunk 2 above)
Comment (Tarang74) [reply client/app/review/create/(components)/(Form)/Form.tsx]: Again apologies, this is unfinalised code, it should not be part of this branch

**Review by Tarang74 — COMMENTED** [review]
Reviewed changes and will create an additional commit

**Review by BrookJeynes — APPROVED** [review]
ship it

**Pull request merged**